src/datasets/domain/models/Dataset.ts

```diff
@@ -0,0 +1,40 @@
+export interface Dataset {
+  id: number;
+  persistentId: string;
+  versionId: number;
+  versionInfo: DatasetVersionInfo;
+  license: DatasetLicense;
+  metadataBlocks: DatasetMetadataBlock[];
+}
+
+export interface DatasetVersionInfo {
+  majorNumber: number;
+  minorNumber: number;
+  state: DatasetVersionState;
```

Comment:
Does this mean that each version has a different state `state: DatasetVersionState;` ? I was treating the state as a global state of the dataset not attached to a version so I have it in the root of the Dataset model, does it make sense?

Reply:
versionState is version specific - a dataset could have v1.0 deaccessioned, v2.0 released, and v2.1 in draft, etc. FWIW: In the current UI, the dataset page is really more a dataset version page in which some things, like the DOI are common. The only element I can think of that is really about the dataset itself is the version tab which shows all versions.

Reply:
Thanks for the clarification! Makes sense now